product_contract/models/sale_order_line.py

```diff
@@ -269,3 +283,53 @@ def _compute_qty_to_invoice(self):
         res = super()._compute_qty_to_invoice()
         self.filtered("product_id.is_contract").update({"qty_to_invoice": 0.0})
         return res
+
+    def _set_contract_line_start_date(self):
+        """Set date start of lines using it's method and the confirmation date."""
+        for line in self:
+            if (
+                line.contract_start_date_method == "manual"
+                or line.recurring_rule_type in ["daily", "weekly", "monthlylastday"]
+            ):
+                continue
+            is_end = "end_" in line.contract_start_date_method
+            today = fields.Date.today()
+            month_period = month = today.month
+            month_nb = MONTH_NB_MAPPING[line.recurring_rule_type]
+            # The period number is started by 0 to be able calculate the month
+            period_number = (month - 1) // month_nb
+            if line.recurring_rule_type == "yearly":
+                month_period = 1
+            elif line.recurring_rule_type != "monthly":
+                # Checking quarterly and semesterly
+                month_period = period_number * month_nb + 1
+            forced_month = 0
+            if line.recurring_rule_type != "monthly":
+                forced_value = int(
+                    line.product_id["force_month_%s" % line.recurring_rule_type]
+                )
+                if forced_value:
+                    # When the selected period is yearly, the period_number field is
+                    # 0, so forced_month will take the value of the forced month set
+                    # on product.
+                    forced_month = month_nb * period_number + forced_value
+            # If forced_month is set, use it, but if it isn't use the month_period
+            start_date = today + relativedelta(
+                day=1, month=forced_month or month_period
+            )
+            if is_end:
+                increment = month_nb if not forced_month else 0
+                start_date = (
+                    start_date
+                    + relativedelta(months=1 + increment)
```

Comment:
Isn't this missing?

```suggestion
                    + relativedelta(months=1 + increment, day=1)
```

Reply:
Already established in the above definition https://github.com/OCA/contract/pull/1120/files#diff-308d8794eb63f4f473a8ce812eba0aacdbdf86a8c0c8f9c3786f1de1baade116R317-R319